src/vos/vos_io.c

```diff
@@ -681,12 +784,17 @@ akey_fetch(struct vos_io_context *ioc, daos_handle_t ak_toh)
 			continue;
 		}
 
-		rc = akey_fetch_recx(toh, &val_epr, &iod->iod_recxs[i], &rsize,
-				     ioc);
-		if (rc != 0) {
-			D_DEBUG(DB_IO, "Failed to fetch index %d: "DF_RC"\n", i,
-				DP_RC(rc));
-			goto out;
+		iod_recx = iod->iod_recxs[i];
+		while (iod_recx.rx_nr > 0) {
+			akey_fetch_recx_get(&iod_recx, shadow, &fetch_recx,
+					    &shadow_ep);
+			rc = akey_fetch_recx(toh, &val_epr, &fetch_recx,
+					     shadow_ep, &rsize, ioc);
```

Comment:
can you use a single epoch?   Will the query never request multiple stripes?

Reply:
it possibly with multiple stripes and different stripe with different epoch.

Reply:
Doesn't that imply an issue here then?   We are using one shadow epoch but consider if we have a request that spans multiple stripes.   This call is using as single epoch.   But lets say we have 2 stripes requested.

Stripe 1 has parity at time 5 and partial overwrite at time 7
Stripe2 has parity at time 8 and prior partial writes at time 6.

What happens here?

Or will they never be within a single request?